nuxeo-features/nuxeo-platform-rendition/nuxeo-platform-rendition-core/src/main/java/org/nuxeo/ecm/platform/rendition/service/RenditionServiceImpl.java

```diff
@@ -384,4 +396,104 @@ protected RenditionDefinition getAvailableRenditionDefinition(DocumentModel doc,
 
         return renditions;
     }
+
+    @Override
+    public void deleteStoredRenditions(String repositoryName) {
+        StoredRenditionsCleaner cleaner = new StoredRenditionsCleaner(repositoryName);
+        cleaner.runUnrestricted();
+    }
+
+    private final class StoredRenditionsCleaner extends UnrestrictedSessionRunner {
+
+        private static final int BATCH_SIZE = 100;
+
+        private StoredRenditionsCleaner(String repositoryName) {
+            super(repositoryName);
+        }
+
+        @Override
+        public void run() {
+            Map<String, List<String>> sourceIdToRenditionRefs = computeLiveDocumentRefsToRenditionRefs();
+            List<String> sourceIds = new ArrayList<>(sourceIdToRenditionRefs.keySet());
+
```

Comment:
If you rework `removeStoredRenditions` as a simple iterative method you don't need to do the copy.